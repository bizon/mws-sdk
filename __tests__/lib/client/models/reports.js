const nock = require('nock')
const MockDate = require('mockdate')

const MWSClient = require('../../../..')

const client = new MWSClient({
  accessKeyId: 'ACCESS_KEY',
  secretAccessKey: 'SECRET_KEY',
  sellerId: 'SELLER_ID',
  mwsToken: 'MWS_TOKEN',
  sellerRegion: 'eu'
})

const apiUrl = `https://${client.settings.region.mwsDomain}`

describe('lib.client.models.reports', () => {
  beforeAll(() => {
    MockDate.set('2019-12-30')
  })

  afterAll(() => {
    MockDate.reset()
    client.reports.clearRestores()
  })

  it('should call GetReport and return a raw string', async () => {
    const {pathname, data} = client.signData('POST', 'Reports', '2009-01-01', {
      Action: 'GetReport',
      ReportId: 'REPORT-1'
    })

    nock(apiUrl)
      .post(pathname, data)
      .reply(200, Buffer.from('Hello world!'))

    const result = await client.reports.getReport({
      reportId: 'REPORT-1',
      format: 'raw'
    })

    expect(result).toEqual('Hello world!')
  })

  it('should call GetReport and return a base64 string', async () => {
    const {pathname, data} = client.signData('POST', 'Reports', '2009-01-01', {
      Action: 'GetReport',
      ReportId: 'REPORT-1'
    })

    nock(apiUrl)
      .post(pathname, data)
      .reply(200, Buffer.from('Hello world!'))

    const result = await client.reports.getReport({
      reportId: 'REPORT-1',
      format: 'base64'
    })

    expect(result).toEqual('SGVsbG8gd29ybGQh')
  })

  it('should call GetReport and return a parsed CSV', async () => {
    const {pathname, data} = client.signData('POST', 'Reports', '2009-01-01', {
      Action: 'GetReport',
      ReportId: 'REPORT-1'
    })

    nock(apiUrl)
      .post(pathname, data)
      .reply(200, Buffer.from('asin,title\nA1,cool product\nA2,other product'))

    const result = await client.reports.getReport({
      reportId: 'REPORT-1'
    })

    expect(result).toMatchSnapshot()
  })
})
