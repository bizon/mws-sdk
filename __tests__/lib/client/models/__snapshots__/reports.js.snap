// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.models.reports should call GetReport and return a parsed CSV 1`] = `
[
  [
    "asin,title",
  ],
  [
    "A1,cool product",
  ],
  [
    "A2,other product",
  ],
]
`;

exports[`lib.client.models.reports should call RequestReport 1`] = `
{
  "completedDate": null,
  "endDate": 2009-02-13T02:10:39.000Z,
  "generatedReportId": null,
  "reportProcessingStatus": "_SUBMITTED_",
  "reportRequestId": "2291326454",
  "reportType": "_GET_MERCHANT_LISTINGS_DATA_",
  "scheduled": false,
  "startDate": 2009-01-21T02:10:39.000Z,
  "startedProcessingDate": null,
  "submittedDate": 2009-02-20T02:10:39.000Z,
}
`;
